### PR TITLE
[SPARK-31271][UI] fix web ui for driver side SQL metrics

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

In https://github.com/apache/spark/pull/23551, we changed the metrics type of driver-side SQL metrics to size/time etc. which comes with max/min/median info.

This doesn't make sense for driver side SQL metrics as they have only one value. It makes the web UI hard to read:
image

This PR updates the SQL metrics UI to only display max/min/median if there are more than one metrics values:
image

### Why are the changes needed?

Makes the UI easier to read

### Does this PR introduce any user-facing change?

no

### How was this patch tested?
manual test

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gengliangwang @sarutak

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 1]
```diff
@@ -180,18 +178,18 @@ object SQLMetrics {
       numberFormat.format(values.sum)
     } else if (metricsType == AVERAGE_METRIC) {
       val validValues = values.filter(_ > 0)
-      val Seq(min, med, max) = {
-        val metric = if (validValues.isEmpty) {
-          val zeros = Seq.fill(3)(0L)
-          zeros.map(v => toNumberFormat(v))
-        } else {
+      if (validValues.length <= 1) {
+        toNumberFormat(validValues.headOption.getOrElse(0))
```
Comment (sarutak) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: It's not obvious that this is for the case of driver so can we leave a comment for the readability?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 2]
```diff
@@ -204,19 +202,19 @@ object SQLMetrics {
       }
 
       val validValues = values.filter(_ >= 0)
-      val Seq(sum, min, med, max) = {
-        val metric = if (validValues.isEmpty) {
-          val zeros = Seq.fill(4)(0L)
-          zeros.map(v => strFormat(v))
-        } else {
+      if (validValues.length <= 1) {
+        strFormat(validValues.headOption.getOrElse(0))
```
Comment (sarutak) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: Same as [here](https://github.com/apache/spark/pull/28037/files#diff-803f475b01acfae1c5c96807c2ea9ddcR182).

**Comment by sarutak** [comment]
@cloud-fan 
Basically LGTM.
But as you suggested, the metrics format for stageId and attemptId can be changed so should we wait for #28039 to be merged.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 3]
```diff
@@ -116,26 +116,23 @@ object SQLMetrics {
     // data size total (min, med, max):
     // 100GB (100MB, 1GB, 10GB)
     val acc = new SQLMetric(SIZE_METRIC, -1)
-    acc.register(sc, name = Some(s"$name total (min, med, max (stageId (attemptId): taskId))"),
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: Could you fix the following in `SQLMetricsTestUtils.scala`?
```scala
    val totalNumBytesMetric = executedNode.metrics.find(
      _.name == "written output total (min, med, max (stageId (attemptId): taskId))").get
```

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 4]
```diff
@@ -204,19 +204,21 @@ object SQLMetrics {
...
+            strFormat(validValues.sum),
+            strFormat(validValues(0)),
             strFormat(validValues(validValues.length / 2)),
-            s"${strFormat(validValues(validValues.length - 1))} $stringMetric")
+            strFormat(validValues(validValues.length - 1)))
         }
-        metric
+        s"total $METRICS_NAME_SUFFIX\n$sum ($min, $med, $max $taskInfo)"
       }
-      s"\n$sum ($min, $med, $max)"
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: In this case, it seems that we didn't have `$taskInfo` previously.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I also verified this locally via web UI. Thank you, all.
Merged to master/3.0.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala (hunk 4 above)
Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: I think it was a mistake. We should have the task info for all the max values.

BTW thanks for catching! I didn't realize it when I was working on this PR ...

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala]: We have, see https://github.com/apache/spark/pull/28037/files#diff-803f475b01acfae1c5c96807c2ea9ddcL215

**Pull request closed without merging**